src/main.rs

```diff
@@ -1,32 +1,70 @@
+use std::io::{Stdin, Stdout, Write};
+
 fn main() {
+    let mut terminal = Terminal::new();
     loop {
-        if generator_todo() {
-            break;
-        }
+        terminal.ask_if_new_todo();
+        let new_todo = terminal.ask_for_new_todo();
+        terminal.show_todo(&new_todo);
     }
 }
 
-fn input() -> String {
-    let mut buf = String::new();
-    std::io::stdin().read_line(&mut buf).unwrap();
-    buf.trim().to_string()
+struct Terminal {
+    stdin: Stdin,
+    stdout: Stdout,
 }
 
-fn generator_todo() -> bool {
-    println!("Você deseja criar um novo TODO? (insira apenas n ou s)");
+impl Terminal {
+    fn new() -> Self {
+        Terminal {
+            stdin: std::io::stdin(),
+            stdout: std::io::stdout(),
+        }
+    }
 
-    let response = input();
+    fn input(&self) -> String {
+        let mut buf = String::new();
+        self.stdin.read_line(&mut buf).unwrap();
+        buf.trim().to_string()
+    }
+
+    fn ask_for_new_todo(&self) -> Todo {
+        println!("Qual será seu novo todo? 💬");
+        let todo_name = self.input();
+        Todo::new(todo_name)
+    }
+    fn ask_if_new_todo(&self) -> bool {
+        println!("Você deseja criar um novo TODO? ⛏  (insira apenas n ou s)");
+        let response = self.input();
 
-    if response == "n" {
-        println!("Tchau brigado");
+        loop {
+    
+            if response == "s" {
+                break;
+            } else if response == "n" {
+                println!("Tchau brigado 🙃");
+                std::process::exit(0);
+            } else {
+                println!("Resposta inválida. Insira apenas 's' para sim ou 'n' para não.");
+                self.ask_if_new_todo();
+                break;
+            }
```

Comment:
Acho que um `match` aqui ficaria mais elegante:

```suggestion
            match response {
                "s" => break,
                "n" => {
                    println!("Tchau brigado 🙃");
                    std::process::exit(0);
                },
                _ => {
                    println!("Resposta inválida. Insira apenas 's' para sim ou 'n' para não.");
                    self.ask_if_new_todo();
                    break;
            }
```

Reply:
De fato fica muito melhor a estrutura